### ISSUE TITLE
How to debug when a streaming site isn't working properly?

### ISSUE THREAD

**Issue by rgoy** [Seji64/SniDust#5]
Hey,

First of all, wanted to say thank you for the amazing work. The docker image you created makes spinning up sniproxy and dnsdist so much easier and seamless.

Wanted to get your thoughts on an issue I am running into where a streaming site (canalplus.com) isn't working properly. Here is the observed behavior:
- I have added 3 top domains I observed to the domains.lst and it seems more successful than having a domain list with really specific ones coming from SmartDNS services such as ControlD or even just the top domains from that list.
- First attempt to watch live TV works perfectly but then switching to a different channel would throw a "Playback error, sorry, a temporary technical problem is preventing access to this programme."
- When getting that error message and switching my DNS configuration to another instance of SniDust on a different VPS, LiveTV works again until it doesn't. Switching back to the previous VPS usually works.
- Never an issue to watch VoD, just live TV that is finicky

**TL;DR** I am having some issues to get a streaming site working and wanted to see if you have ideas on how I could debug that?

Thanks so much!

**Comment by Seji64** [comment]
Hi, thanks for the applause :-)

unfortunately i don't really have a tip how to troubleshoot that kind of issue. The only option that comes to my mind is to look at the network traffic (Browser Dev Tools) to see which domains are called and which domain may cause your issue. Probably it is some kind of CDN which is need to be blocked.

**Comment by rgoy** [comment]
Is there a way to see which domains are being called by the clients and which one are being redirected by sniproxy/dnsdist?

Would you recommend blocking only the top domains (i.e. canal-plus.com) or to include subdomains as much as possible (i.e. secure-webtv-static.canal-plus.com)? I see a combination of both in your domains.lst

**Comment by Seji64** [comment]
> Is there a way to see which domains are being called by the clients and which one are being redirected by sniproxy/dnsdist?

if you lookup the specific domain with nslookup/dig and the answer is your VPS IP then its handled by SniDust. Other method will be to enable Query loggin in dnsdist.

> Would you recommend blocking only the top domains (i.e. canal-plus.com) or to include subdomains as much as possible (i.e. secure-webtv-static.canal-plus.com)? I see a combination of both in your domains.lst

if its a 'static' subdomain i would use this. but sometimes its like 00.cdn.blah.net, 01.cdn.blah.net and so on. then it is hard guess which will be used. In this case I would try to block cdn.blah.net.
